test/test-options-cluster.js

```diff
@@ -0,0 +1,42 @@
+// Copyright IBM Corp. 2015. All Rights Reserved.
+// Node module: strong-supervisor
+// This file is licensed under the Artistic License 2.0.
+// License text available at https://opensource.org/licenses/Artistic-2.0
+
+var CPUS = require('os').cpus().length;
+var options = require('../lib/options');
+var tap = require('tap');
+
+tap.test('cluster option validation', function(t) {
+  t.test('valid values', function(t) {
+    check(['node', 'sl-run', '--cluster=10'], 10);
+    check(['node', 'sl-run', '--cluster', '-10'], -10);
+    check(['node', 'sl-run', '--cluster', 'CPU'], CPUS);
+    check(['node', 'sl-run', '--cluster=cpu'], CPUS);
+    t.end();
+
+    function check(argv, expect) {
+      var o = options.parse(argv);
+      t.equal(o.cluster.size, expect);
```

Comment:
In this case, o.cluster is always there, but I have run into problems like this. Basically, one tests fail, they tend to continue to fail more and more badly. Pretty much any test with multiple asserts can be prone to this, once an early assert fails, then the test state is indeterminate. Its already failed, the follow-on assertions are possibly syntactically invalid at this point, in that their code may fail with attempts to acess undefined objects, or call methods that don't exist, and worse, any cleanup that the test does may fail, too. Sometimes I've written defensive code around this for tests that are prone to regression, but its an odd sort of defensive coding... particularly because the defensive code may never run if the test never fails again. And that means its code that effectively never gets covered or tested, but its test code, quis custodiet ipsos custodes?

I suspect there is some kind of lodash pick function that could be used instead of the `.. && ...` idiom: `t.equal(_.xxx(o, 'cluster', 'size'), expect)`, your second example is pretty readable, too.